samcli/cli/cli_config_file.py

```diff
@@ -133,7 +133,11 @@ def configuration_callback(cmd_name, option_name, saved_callback, provider, ctx,
     config_dir = getattr(ctx, "samconfig_dir", None) or os.getcwd()
     # If --config-file is an absolute path, use it, if not, start from config_dir
     config_file_path = config_file if os.path.isabs(config_file) else os.path.join(config_dir, config_file)
-    if config_file and config_file != DEFAULT_CONFIG_FILE_NAME and not Path(config_file_path).absolute().is_file():
+    if (
+        config_file
+        and config_file != DEFAULT_CONFIG_FILE_NAME
+        and not (Path(config_file_path).absolute().is_file() or Path(config_file_path).absolute().is_fifo())
```

Comment:
On the suggestion to refactor to `open()`, I'll have a look and I'll come back to you.

On testing on Windows, I'll try to verify how `is_fifo()` works there.

Reply:
I looked a bit more into the Windows case. Windows supports named pipes and `is_fifo` returns `True` when it's called with one (I created the named pipe with `pywin32`'s `win32pipe.CreateNamedPipe`).

Also, I made an attempt at refactoring my change according to your suggestion:
* I had to use `os.open` instead of the built-in `open` as I had to pass `os.O_NONBLOCK` when opening the pipe
* I kept the `config_file != DEFAULT_CONFIG_FILE_NAME` check for now, as omitting it requires a lot of changes in existing tests (I'm ok to do that if we eventually go with this approach)
* I wanted to make the `except` clause more specific, but I was unable to determine the exact exceptions that can be raised across the supported operating systems

What do you think?

Reply:
> I had to use os.open instead of the built-in open as I had to pass os.O_NONBLOCK when opening the pipe

Unfortunately, `os.O_NONBLOCK` is not supported on Windows. While I don't mind investigating the differences between Linux and Windows, I believe the resulting solution will end up being too complex for the value it's going to add. Therefore, I'll return to my initial approach, enhancing the "if" clause.

Reply:
Thanks for the deep dive. What it is now looks good enough to me.